controller/authorize_blackbox_test.go

```diff
@@ -160,15 +178,21 @@ func (rest *TestAuthorizeREST) TestAuthorizeCallbackOK() {
 	require.True(t, strings.HasPrefix(locationString, redirectURI))
 	require.Nil(t, err)
 
-	allQueryParameters = locationUrl.Query()
-
-	require.NotNil(t, allQueryParameters)
-	require.NotNil(t, allQueryParameters["state"][0])
-	require.Equal(t, returnedState, allQueryParameters["state"][0])
-	require.NotNil(t, allQueryParameters["code"][0])
-	require.Equal(t, code, allQueryParameters["code"][0])
+	if responseMode == nil || *responseMode != "fragment" {
+		require.NotNil(t, locationUrl.RawQuery)
+		allQueryParameters = locationUrl.Query()
+
+		require.NotNil(t, allQueryParameters)
+		require.NotNil(t, allQueryParameters["state"][0])
+		require.Equal(t, returnedState, allQueryParameters["state"][0])
+		require.NotNil(t, allQueryParameters["code"][0])
+		require.Equal(t, code, allQueryParameters["code"][0])
+	} else {
+		require.NotNil(t, locationUrl.Fragment)
+		require.True(t, strings.HasPrefix(locationUrl.Fragment, "code"))
+		require.Contains(t, locationUrl.Fragment, "state")
```

Comment:
I think I already said it before.. Do check the actual returned code in the fragment.